### ISSUE TITLE
.NET Core 2 support?

### ISSUE THREAD

**Issue by tiwari-abhishek** [Azure/autorest#2813]
I tried to add autorest to a .NET Core 2 project. on using it from Nuget Package Manager Console it throws error about not having .net Core 1.0.4.

I am on .netcore 2.0 and don't have any other versions installed. Is there a plan to support latest .net core versions or the only way is to install an older .net version to get this to work?

    PM> autorest
    autorest : It was not possible to find any compatible framework version
    At line:1 char:1
    + autorest
    + ~~~~~~~~
    + CategoryInfo          : NotSpecified: (It was not poss...amework version:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
     The specified framework 'Microsoft.NETCore.App', version '1.0.4' was not found.
      - Check application dependencies and target a framework version installed at:
       \
      - Alternatively, install the framework version '1.0.4'.

**Comment by fearthecowboy** [comment]
What version of AutoRest are you using? It sounds like an older version.

Get the latest version via npm:

``` powershell
npm install -g autorest 

```

**Comment by tiwari-abhishek** [comment]
I tried using `npm install -g autorest` but it only gave me a folder containing a few files and so it doesn't run
![image](https://user-images.githubusercontent.com/12059624/35288289-e8ac00b6-0029-11e8-8d31-259a93a2aea1.png)

On running autorest it just hangs in the console window without errors:
![image](https://user-images.githubusercontent.com/12059624/35290559-3f816826-0030-11e8-9b52-8fb3e3b30683.png)

When using the autorest zip available at https://github.com/Azure/autorest/releases
I am able to run the same command and generate the code.

Also I notice the Zip contains way too many packages and files than what I get on running the `npm install autorest -g`. Its a 90MB download.

Is the npm installation broken or is it some extra step that I am missing?

**Comment by fearthecowboy** [comment]
You're missing an equals sign after the `--input-file` .  I think it's trying to treat the URL as a configuration instead of input. Although, it shouldn't *hang* there. Hmm. That's gotta be a bug. 

try : 
``` powershell
autorest --input-file=http://localhost:53770/swagger/v1/swagger.json --csharp ...
```

The zip file contains a snapshot of all the autorest extension packages. On a normal install via `npm`, it doesn't drag down all the autorest extensions all at once, it installs them as you use them (ie, when you say `--csharp`, it'll grab the csharp extension and install it into `$HOME/.autorest/...`

**Comment by tiwari-abhishek** [comment]
sorry that was a typo. I did actually use = after --input-file but it doesnt do anything. same as the screenshot.

**Comment by fearthecowboy** [comment]
Let's try a couple more things:

``` powershell
# let's grab the preview version, just to pick up the very latest fixes 
npm install -g autorest@preview 

# reset all the plugins, make sure nothing is broken.
autorest --reset 

# show what's installed
autorest --info 

# run it with verbose and debug and see what it says.
autorest  --verbose --debug --input-file=http://localhost:53770/swagger/v1/swagger.json --csharp ...

```

If the swagger is very large, it may take a few minutes; part of the process uses roslyn to simplify/clean the code, and that's not the fastest operation.

If this still fails, can you post the swagger file (or send it to me at `garretts`@`microsoft.com` ) and I can repro it here?

**Comment by tiwari-abhishek** [comment]
Thank you so much!
To confirm what was already present on my machine, I ran autorest --info before installing preview and it said no extensions installed.
After running the `npm install -g autorest@preview` followed by `autorest --reset` like you mentioned, I was able to see 1 plugin installed (autorest-core).

And the code generation worked fine after that, pulling down autorest.csharp like you said. So thank you again!
